packages/nimble-components/src/table/specs/table-row-grouping-hld.md

```diff
@@ -0,0 +1,238 @@
+            <${DesignSystem.tagFor(TableGroupRow)}
+                :groupRowValue="${(x, c) => c.parent.tableData[x.index]?.groupRowValue}"
+                ...
+                @on-row-group-expand-toggle="${(x, c) => c.parent.toggleRowExpanded(x.index)}"
+                >
+            </${DesignSystem.tagFor(TableGroupRow)}>
+        `)}
+        ${when((x, c) => !(c.parent as Table).tableData[x.index]?.isGrouped, html<VirtualItem, Table>`
+            <${DesignSystem.tagFor(TableRow)}
+    ...
+```
+
+The one change necessary to support this properly is to change the virtualizer to use the current length of `getRowModel.rows()` for the `count` instead of the [current length of the table's data](https://github.com/ni/nimble/blob/f60dedf600147b16916f876b749a4e019c2a6308/packages/nimble-components/src/table/models/virtualizer.ts#L82) (see [prototype](https://github.com/ni/nimble/blob/5cd05ab733b00ebb33e44d53c7be30c980fa3398/packages/nimble-components/src/table/models/virtualizer.ts#L87)).
+
+### Column header grouping visual
+
+Ultimately, when a column has set itself to be grouped, the design calls for that to be represented with a visual in the header for that column. This will be handled in a similar fashion to how we represent sort direction, by providing a new `is-grouped` attribute on `TableHeader` to determine whether to show the appropriate icon.
+
+### Interfacing with TanStack
+
+As shown in the above example, the `TableGroupRow` will provide a button to control expand/collapse behavior. TanStack handles the modeling for which rows are expanded or collapsed (in addition to which rows are groups), and thus must be notified of an expand/collapse action. This will be accomplished through raising an event (`group-expand-toggle` in the example above). The `Table` will handle the event, and call `stopPropogation` on it, as there is not an immediate need to leak this into the public event API of the `Table`.
+
+Setting TanStack up to handle both row grouping and expanded state is similar to what is in place for sorting. One notable difference is that we will need to register a handler for TanStack's `onExpandedChange` handler, as we will be working through TanStack to update this state, and the `Table` will need a notification in order to re-render the rows. It would look similar to the following:
+
+Table index.ts
+
+```
+public constructor() {
+    super();
+    this.options = {
+        data: [],
+        onStateChange: (_: TanStackUpdater<TanStackTableState>) => {},
+        onExpandedChange: this.handleExpandedChange,
+        ...
+
+```
+
+### Validation
+
+The table will be invalid if it contains two columns with the same `group-index`. If a duplicate `group-index` is detected, the rows in the table will not render, and the table's validity object will have `duplicateColumnGroupIndex` set to `true`.
+
+## Alternative Implementations / Designs
```

Comment:
It would be good to call out the rationale for our decision to group by operand data value rather than rendered value.